flex_container_orchestrator/domain/lead_time_aggregator.py

```diff
@@ -0,0 +1,319 @@
+    time_increment = time_settings['tincr']
+    run_frequency = time_settings['tfreq']
+
+    all_input_forecasts_set = set()
+    all_input_forecasts = []
+    all_flexpart_leadtimes = []
+
+    for start_time in start_times:
+        lead_times = [start_time + datetime.timedelta(hours=i) for i in range(0, time_delta, time_increment)]
+        input_forecasts = [generate_forecast_label(lt, run_frequency) for lt in lead_times]
+
+        all_input_forecasts_set.update(input_forecasts)
+        all_input_forecasts.append(input_forecasts)
+        all_flexpart_leadtimes.append(lead_times)
+
+    return all_input_forecasts, all_flexpart_leadtimes, all_input_forecasts_set
+
+
+
+def strip_lead_time(forecast: str) -> datetime.datetime:
+    """
+    Strip lead time from forecast string.
+
+    Args:
+        forecast (str): Forecast reference time with lead time.
+
+    Returns:
+        datetime.datetime: Forecast datetime without lead time.
+    """
+    return datetime.datetime.strptime(forecast[:-2], "%Y%m%d%H%M")
+
+
+def create_flexpart_configs(
+    all_flexpart_leadtimes: list[list[datetime.datetime]],
+    all_input_forecasts: list[list[str]],
+    processed_forecasts: set[str],
+) -> list[dict]:
+    """
+    Create Flexpart input configurations based on processed forecasts.
+
+    Args:
+        - all_input_forecasts: A nested list where each sublist contains forecast labels
+            for each Flexpart run in the format "{reference_time}_{step}".
+        - all_flexpart_leadtimes: A nested list where each sublist contains datetime objects
+            representing the lead times for each Flexpart run.
+        - processed_forecasts (set of str): Set of forecasts marked as processed and retrieved from the DB for
+            which have the same reference times as the forecasts needed for the Flexpart simulation
+
+    Returns:
+        list of dict: List of Flexpart configuration dictionaries. If no valid
+            configurations can be created, an empty list is returned.
+    """
+    configs = []
+    for run_index, input_forecasts_for_run in enumerate(all_input_forecasts):
+        if all(forecast in processed_forecasts for forecast in input_forecasts_for_run):
+            config = define_config(all_flexpart_leadtimes[run_index][0], all_flexpart_leadtimes[run_index][-1])
+            configs.append(config)
+    return configs
+
+
+def run_aggregator(date: str, time: str, step: int) -> list[dict]:
+    """
+    Checks if Flexpart can be launched with the processed new lead time and prepares input configurations.
+
+    Args:
+        date (str): The forecast reference date in YYYYMMDD format.
+        time (str): The forecast reference time in HH format.
+        step (int): The lead time in hours.
+
+    Returns:
+        list[dict]: List of configuration dictionaries for Flexpart.
+    """
+    time_settings = get_time_settings(CONFIG)
+
+    DBtable = os.path.join(CONFIG.main.db.path, CONFIG.main.db.name)
```

Comment:
```suggestion
    db_path = os.path.join(CONFIG.main.db.path, CONFIG.main.db.name)
``` 
see https://peps.python.org/pep-0008/#function-and-variable-names